ChatSecure/Classes/Controllers/YapDatabaseTransaction+ChatSecure.swift

```diff
@@ -107,6 +107,29 @@ public extension YapDatabaseReadTransaction {
         
         return result
     }
+    
+    //#865
+    @objc public func allMediaMessagesForThread(_ thread:OTRThreadOwner) -> [OTRMessageProtocol] {
+        guard let indexTransaction = self.ext(OTRMessagesSecondaryIndex) as? YapDatabaseSecondaryIndexTransaction else {
+            return []
+        }
+        let queryString = "Where \(OTRYapDatabaseMessageThreadIdSecondaryIndexColumnName) == ?"
+        let query = YapDatabaseQuery(string: queryString, parameters: [thread.threadIdentifier])
+        var result = [OTRMessageProtocol]()
+        let success = indexTransaction.enumerateKeysAndObjects(matching: query) { (collection, key, object, stop) in
```

Comment:
Instead of iterating over every message, it would be a faster query to add `messageMediaItemKey` to the secondary index columns (see setupSecondaryIndexes in OTRDatabaseManager).

Reply:
I'm struggling with this one, can't quite figure out how this is supposed to work. Our input parameter to this function is OTRThreadOwner, and we are checking all the messages in a thread to see if they are media messages. Would the new secondary column then be in setupMessageSecondaryIndexes instead of setupSecondaryIndexes?

Also, we are just checking the existence of mediaMessages here, so should we just query for the existence of data in this column? Something like:

let queryString = "Where \(OTRYapDatabaseMessageThreadIdSecondaryIndexColumnName) == ? AND \(OTRYapDatabaseMessageMediaItemKeySecondaryIndexColumnName) != NULL"